File: frontend/packages/openmates-cli/examples/chat-and-app-skills.mjs
#!/usr/bin/env node
/*
 * OpenMates npm SDK example: chats and app skills.
 *
 * Run after building the package:
 *   OPENMATES_API_KEY=sk-api-... node examples/chat-and-app-skills.mjs
 *
 * This uses real API requests. It lists encrypted account chats, loads the first
 * chat when available, and runs a deterministic math app skill.
 */

import { OpenMates } from "../dist/index.js";

const client = new OpenMates({
  apiKey: process.env.OPENMATES_API_KEY,
  apiUrl: process.env.OPENMATES_API_URL,
});

const chats = await client.chats.list({ limit: 10 });
const firstChat = chats[0]?.id ? await client.chats.load(String(chats[0].id)) : null;
const calculation = await client.apps.math.calculate({
  title: "SDK example calculation",
  expression: "2 + 2",
});

console.log(JSON.stringify({
  chats: chats.map((chat) => ({
    id: chat.id,
    title: chat.title ?? null,
    category: chat.category ?? null,
  })),
  loadedChat: firstChat
    ? {
        id: firstChat.chat?.id ?? null,
        messageCount: Array.isArray(firstChat.messages) ? firstChat.messages.length : 0,
        embedCount: Array.isArray(firstChat.embeds) ? firstChat.embeds.length : 0,
      }
    : null,
  calculation,
}, null, 2));

File: packages/openmates-python/examples/chat_and_app_skills.py
#!/usr/bin/env python3
"""OpenMates Python SDK example: chats and app skills.

Run from the repository root or installed package environment:
  OPENMATES_API_KEY=sk-api-... PYTHONPATH=packages/openmates-python python3 packages/openmates-python/examples/chat_and_app_skills.py

This uses real API requests. It lists encrypted account chats, loads the first
chat when available, and runs a deterministic math app skill.
"""

from __future__ import annotations

import json
import os

from openmates import OpenMates


client = OpenMates(
    api_key=os.getenv("OPENMATES_API_KEY"),
    api_url=os.getenv("OPENMATES_API_URL", "https://api.openmates.org"),
)

chats = client.chats.list(limit=10)
first_chat = client.chats.load(str(chats[0]["id"])) if chats and chats[0].get("id") else None
calculation = client.apps.math.calculate({"title": "SDK example calculation", "expression": "3 + 4"})

print(json.dumps({
    "chats": [
        {
            "id": chat.get("id"),
            "title": chat.get("title"),
            "category": chat.get("category"),
        }
        for chat in chats
    ],
    "loadedChat": {
        "id": first_chat.get("chat", {}).get("id"),
        "messageCount": len(first_chat.get("messages", [])),
        "embedCount": len(first_chat.get("embeds", [])),
    } if isinstance(first_chat, dict) else None,
    "calculation": calculation,
}, indent=2))
